File: js_sdk/src/model/ShareCancelBody.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';
import {DmpcustomAudiencefileuploadContextInfo} from './DmpcustomAudiencefileuploadContextInfo.js';

/**
 * The ShareCancelBody model module.
 * @module model/ShareCancelBody
 * @version 0.1.3
 */
export class ShareCancelBody {
  /**
   * Constructs a new <code>ShareCancelBody</code>.
   * @alias module:model/ShareCancelBody
   * @class
   * @param advertiser_id {String} The advertiser ID that performed the operation.
   * @param custom_audience_id {String} The audience ID that you want to stop sharing.
   * @param shared_advertiser_id {String} The advertiser ID you want to stop sharing with. You only need to pass this field if you want to stop sharing audiences with an advertiser.
   */
  constructor(advertiser_id, custom_audience_id, shared_advertiser_id) {
    this.advertiser_id = advertiser_id;
    this.custom_audience_id = custom_audience_id;
    this.shared_advertiser_id = shared_advertiser_id;
  }

  /**
   * Constructs a <code>ShareCancelBody</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/ShareCancelBody} obj Optional instance to populate.
   * @return {module:model/ShareCancelBody} The populated <code>ShareCancelBody</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new ShareCancelBody();
      if (data.hasOwnProperty('advertiser_id'))
        obj.advertiser_id = ApiClient.convertToType(data['advertiser_id'], 'String');
      if (data.hasOwnProperty('context_info'))
        obj.context_info = DmpcustomAudiencefileuploadContextInfo.constructFromObject(data['context_info']);
      if (data.hasOwnProperty('custom_audience_id'))
        obj.custom_audience_id = ApiClient.convertToType(data['custom_audience_id'], 'String');
      if (data.hasOwnProperty('shared_advertiser_id'))
        obj.shared_advertiser_id = ApiClient.convertToType(data['shared_advertiser_id'], 'String');
    }
    return obj;
  }
}

/**
 * The advertiser ID that performed the operation.
 * @member {String} advertiser_id
 */
ShareCancelBody.prototype.advertiser_id = undefined;

/**
 * @member {module:model/DmpcustomAudiencefileuploadContextInfo} context_info
 */
ShareCancelBody.prototype.context_info = undefined;

/**
 * The audience ID that you want to stop sharing.
 * @member {String} custom_audience_id
 */
ShareCancelBody.prototype.custom_audience_id = undefined;

/**
 * The advertiser ID you want to stop sharing with. You only need to pass this field if you want to stop sharing audiences with an advertiser.
 * @member {String} shared_advertiser_id
 */
ShareCancelBody.prototype.shared_advertiser_id = undefined;

File: java_sdk/src/main/java/business_api_client/ShareCancelBody.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import business_api_client.DmpcustomAudiencefileuploadContextInfo;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
/**
 * ShareCancelBody
 */

@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2025-03-21T10:49:20.864324-07:00[America/Los_Angeles]")
public class ShareCancelBody {
  @JsonProperty("advertiser_id")
  private String advertiserId = null;

  @JsonProperty("context_info")
  private DmpcustomAudiencefileuploadContextInfo contextInfo = null;

  @JsonProperty("custom_audience_id")
  private String customAudienceId = null;

  @JsonProperty("shared_advertiser_id")
  private String sharedAdvertiserId = null;

  public ShareCancelBody advertiserId(String advertiserId) {
    this.advertiserId = advertiserId;
    return this;
  }

   /**
   * The advertiser ID that performed the operation.
   * @return advertiserId
  **/
  @Schema(required = true, description = "The advertiser ID that performed the operation.")
  public String getAdvertiserId() {
    return advertiserId;
  }

  public void setAdvertiserId(String advertiserId) {
    this.advertiserId = advertiserId;
  }

  public ShareCancelBody contextInfo(DmpcustomAudiencefileuploadContextInfo contextInfo) {
    this.contextInfo = contextInfo;
    return this;
  }

   /**
   * Get contextInfo
   * @return contextInfo
  **/
  @Schema(description = "")
  public DmpcustomAudiencefileuploadContextInfo getContextInfo() {
    return contextInfo;
  }

  public void setContextInfo(DmpcustomAudiencefileuploadContextInfo contextInfo) {
    this.contextInfo = contextInfo;
  }

  public ShareCancelBody customAudienceId(String customAudienceId) {
    this.customAudienceId = customAudienceId;
    return this;
  }

   /**
   * The audience ID that you want to stop sharing.
   * @return customAudienceId
  **/
  @Schema(required = true, description = "The audience ID that you want to stop sharing.")
  public String getCustomAudienceId() {
    return customAudienceId;
  }

  public void setCustomAudienceId(String customAudienceId) {
    this.customAudienceId = customAudienceId;
  }

  public ShareCancelBody sharedAdvertiserId(String sharedAdvertiserId) {
    this.sharedAdvertiserId = sharedAdvertiserId;
    return this;
  }

   /**
   * The advertiser ID you want to stop sharing with. You only need to pass this field if you want to stop sharing audiences with an advertiser.
   * @return sharedAdvertiserId
  **/
  @Schema(required = true, description = "The advertiser ID you want to stop sharing with. You only need to pass this field if you want to stop sharing audiences with an advertiser.")
  public String getSharedAdvertiserId() {
    return sharedAdvertiserId;
  }

  public void setSharedAdvertiserId(String sharedAdvertiserId) {
    this.sharedAdvertiserId = sharedAdvertiserId;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    ShareCancelBody shareCancelBody = (ShareCancelBody) o;
    return Objects.equals(this.advertiserId, shareCancelBody.advertiserId) &&
        Objects.equals(this.contextInfo, shareCancelBody.contextInfo) &&
        Objects.equals(this.customAudienceId, shareCancelBody.customAudienceId) &&
        Objects.equals(this.sharedAdvertiserId, shareCancelBody.sharedAdvertiserId);
  }

  @Override
  public int hashCode() {
    return Objects.hash(advertiserId, contextInfo, customAudienceId, sharedAdvertiserId);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class ShareCancelBody {\n");
    
    sb.append("    advertiserId: ").append(toIndentedString(advertiserId)).append("\n");
    sb.append("    contextInfo: ").append(toIndentedString(contextInfo)).append("\n");
    sb.append("    customAudienceId: ").append(toIndentedString(customAudienceId)).append("\n");
    sb.append("    sharedAdvertiserId: ").append(toIndentedString(sharedAdvertiserId)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}
